streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -775,26 +806,16 @@ private void completeTaskCloseClean(final Task task) {
     void shutdown(final boolean clean) {
         final AtomicReference<RuntimeException> firstException = new AtomicReference<>(null);
 
-        final Set<Task> tasksToCloseDirty = new HashSet<>();
```

Comment:
No actual changes here, just pulled the cleanup of tasks out into a separate new `#closeAndCleanUpTasks` method so we can call that on tasks from removed topologies